src/NServiceBus.Metrics.ServiceControl/ReportingFeature.cs

```diff
@@ -1,342 +1,304 @@
-﻿namespace NServiceBus.Metrics.ServiceControl
+﻿namespace NServiceBus.Metrics.ServiceControl;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Features;
+using Hosting;
+using Logging;
+using MessageMutator;
+using Microsoft.Extensions.DependencyInjection;
+using Performance.TimeToBeReceived;
+using Routing;
+using ServiceControlReporting;
+using Support;
+using Transport;
+
+sealed class ReportingFeature : Feature
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Features;
-    using Hosting;
-    using Logging;
-    using MessageMutator;
-    using Microsoft.Extensions.DependencyInjection;
-    using Performance.TimeToBeReceived;
-    using Routing;
-    using ServiceControlReporting;
-    using Support;
-    using Transport;
-
-    sealed class ReportingFeature : Feature
+    public ReportingFeature() => DependsOn<MetricsFeature>();
+
+    protected override void Setup(FeatureConfigurationContext context)
     {
-        public ReportingFeature()
-        {
-            EnableByDefault();
-            DependsOn<MetricsFeature>();
-            Prerequisite(ctx =>
-            {
-                var options = ctx.Settings.GetOrDefault<MetricsOptions>();
-                if (options == null)
-                {
-                    return false;
-                }
+        var settings = context.Settings;
+        var options = settings.Get<MetricsOptions>();
+        var endpointName = settings.EndpointName();
+
+        var metrics = new Dictionary<string, Tuple<RingBuffer, TaggedLongValueWriterV1>>();
 
-                var reportingOptions = ReportingOptions.Get(options);
-                var address = reportingOptions.ServiceControlMetricsAddress;
-                return string.IsNullOrEmpty(address) == false;
-            }, $"Reporting is enabled by calling '{nameof(MetricsOptionsExtensions.SendMetricDataToServiceControl)}'");
+        void RegisterDuration(IDurationProbe probe)
+        {
+            var buffer = new RingBuffer();
+            var writer = new TaggedLongValueWriterV1();
+            var name = probe.Name.Replace(" ", "");
 
-            Defaults(settings =>
+            probe.Register((ref e) =>
             {
-                var options = settings.GetOrDefault<MetricsOptions>();
-                if (options != null)
-                {
-                    var reportingOptions = ReportingOptions.Get(options);
-                    settings.Set("NServiceBus.Metrics.ServiceControl.MetricsAddress", reportingOptions.ServiceControlMetricsAddress);
```

Comment:
This is not used anywhere https://github.com/search?q=org%3AParticular+NServiceBus.Metrics.ServiceControl.MetricsAddress&type=code